api/internal/localizer/localizer.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"log"
+	"path/filepath"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	plgnsLdr "sigs.k8s.io/kustomize/api/internal/plugins/loader"
+	"sigs.k8s.io/kustomize/api/internal/target"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/yaml"
+)
+
+// Localizer encapsulates all state needed to localize the root at ldr.
+type Localizer struct {
+	fSys filesys.FileSystem
+
+	// kusttarget fields
+	validator ifc.Validator
+	rFactory  *resmap.Factory
+	pLdr      *plgnsLdr.Loader
+
+	// should be LocLoader
+	ldr ifc.Loader
+
+	// destination directory in newDir that mirrors ldr's current root.
+	dst filesys.ConfirmedDir
+}
+
+// NewLocalizer is the factory method for Localizer
+func NewLocalizer(ldr *LocLoader, validator ifc.Validator, rFactory *resmap.Factory, pLdr *plgnsLdr.Loader) (*Localizer, error) {
```

Comment:
I'd expect `NewLocalizer` only returns Localizer, not error.

can we change the MkdirAll a log fatal, or you plan to further handle the  wrapped error later on?

e.g.
```
func NewLocalizer(ldr *LocLoader, validator ifc.Validator, rFactory *resmap.Factory, pLdr *plgnsLdr.Loader) *Localizer {
	toDst, err := filepath.Rel(ldr.args.Scope.String(), ldr.Root())
	if err != nil {
		log.Fatalf("cannot find path from directory %q to %q inside directory: %s", ldr.args.Scope.String(),
			ldr.Root(), err.Error())
	}
	dst := ldr.args.NewDir.Join(toDst)
	if err = ldr.fSys.MkdirAll(dst); err != nil {
		log.Fatalf("unable to create directory in localize destination: %s", err.Error())
	}
	return &Localizer{
		fSys:      ldr.fSys,
		validator: validator,
		rFactory:  rFactory,
		pLdr:      pLdr,
		ldr:       ldr,
		dst:       filesys.ConfirmedDir(dst)
	}
}
```

Reply:
I return an error because the upper layer can clean up the destination directory if there are any errors here. Lmk if you still want me to change the `error` to `log.Fatal`.